test/conftest.py

```diff
@@ -73,6 +72,7 @@ def kafka_consumer_factory(kafka_broker, topic, request):
     def factory(**kafka_consumer_params):
         params = {} if kafka_consumer_params is None else kafka_consumer_params.copy()
         params.setdefault('client_id', 'consumer_%s' % (request.node.name,))
+        params.setdefault('auto_offset_reset', 'earliest')
```

Comment:
do we really want this? the default in the consumers is normally `'latest'` and I'm a little concerned that this sets a default for test purposes that does not match normal behavior.

Reply:
It gave me problems, at least. In tests, you set up by sending messages before consuming. So having a `latest` setting require you to connect before set up, which is not what you want in several places.

Reply:
Understood.

For my in-house tests of our internal wrapper, I did the connect-then-send messages and while slightly more verbose, it hasn't been too onerous.

That said, I'm fine with going this route instead.